catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/rtl/ShowTextBrickSupportNonLatinStringAndNumberTest.java

```diff
@@ -0,0 +1,244 @@
+		printNonLatinStringOnStage(String.format(new Locale("ta"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowTeluguNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("te"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowThaiNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("th"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowMalayalamNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("ml"), "%f", zeroToNine));
+	}
+
+	private Script createProjectAndGetStartScriptWithImages(String projectName) {
+		Project project = new Project(null, projectName);
+		Sprite sprite = new Sprite("testSprite");
+		Script script = new StartScript();
+
+		sprite.addScript(script);
+
+		script.addBrick(new SetVariableBrick());
+		script.addBrick(new ShowTextBrick());
+
+		project.getDefaultScene().addSprite(sprite);
+
+		ProjectManager.getInstance().setProject(project);
+		ProjectManager.getInstance().setCurrentSprite(sprite);
+		return script;
+	}
+
+	private void printNonLatinStringOnStage(String variableValue) {
+		final String variableName = "testVariable";
+
+		onView(withId(R.id.button_play))
+				.perform(click());
+		onView(isRoot())
+				.perform(CustomActions.wait(1500));
+
+		stageListener.makeManualScreenshot();
+		pressBack();
+		pressBack();
+
+		String filePath = Environment.getExternalStorageDirectory().getAbsolutePath();
+		File file1 = new File(filePath, "/Pocket Code uiTest/showTextBrickTest1/Scene 1/manual_screenshot.png");
+		Bitmap before = null;
+		try {
+			before = BitmapFactory.decodeStream(new FileInputStream(file1));
+		} catch (FileNotFoundException e) {
+			Log.d(ShowTextBrickSupportNonLatinStringAndNumberTest.class.getSimpleName(), "printNonLatinStringOnStage: " + e
+					.getMessage());
+		}
+
+		onView(withId(R.id.set_variable_spinner))
+				.perform(click());
+		onView(withId(R.id.dialog_formula_editor_data_name_edit_text))
+				.perform(typeText(variableName));
+		onView(withText(R.string.ok))
+				.perform(click());
+
+		onView(withId(R.id.show_variable_spinner))
+				.perform(click());
+		onView(withText(variableName))
+				.perform(click());
+
+		onView(withId(R.id.brick_set_variable_edit_text))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_string))
+				.perform(click());
+		onView(withText(R.string.formula_editor_new_string_name))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.formula_editor_string_name_edit_text))
+				.perform(replaceText(variableValue));
+		onView(withText(R.string.ok))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_ok))
+				.perform(click());
+
+		onView(withId(R.id.formula_editor_edit_field))
+				.check(matches(withText("'" + variableValue + "' ")));
+		onView(withId(R.id.button_play))
+				.perform(click());
+		onView(isRoot())
+				.perform(CustomActions.wait(1500));
+
+		stageListener.makeManualScreenshot();
```

Comment:
perfect would be if you could solve this without write / read actions on the sd card, eg, by doing what makeManualScreenshot does, but without saving, so calling ScreenUtils.getFrameBufferPixels(screenshotX, screenshotY, screenshotWidth, screenshotHeight,	true); directly and using the contents. (if that is possible...)